store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/AccumuloTestClusterManager.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016-2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.accumulostore;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.SystemPermission;
+import org.apache.accumulo.minicluster.MiniAccumuloCluster;
+import org.apache.accumulo.minicluster.MiniAccumuloConfig;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.gaffer.store.StoreProperties;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+
+public class AccumuloTestClusterManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AccumuloTestClusterManager.class);
+    private static final String BASE_DIRECTORY = "miniAccumuloStoreTest-";
+    public static final String ROOTPW = "password";
+    private MiniAccumuloCluster miniAccumuloCluster = null;
+    private AccumuloProperties accumuloProperties;
+
+    public AccumuloTestClusterManager(AccumuloProperties inputProperties) {
+        // Check if we need a mini cluster set up from reading the properties
+        accumuloProperties = inputProperties;
+        final String storeClass = accumuloProperties.getStoreClass();
+        if (null == storeClass) {
+            Class currentClass = new Object() { }.getClass().getEnclosingClass();
+            throw new IllegalArgumentException(currentClass.getName() +
+                    ": The Store class name was not found in the store properties for key: " +
+                    StoreProperties.STORE_CLASS);
+        }
+        if (storeClass.equals("uk.gov.gchq.gaffer.accumulostore.SingleUseMiniAccumuloStore") ||
```

Comment:
It might be better to use `SingleUseMiniAccumuloStore.class.getName()` rather than hardcoding this. (same for the line below)

Reply:
Good point. I will attend to it.